src/rules/deprecation.ts

```diff
@@ -128,7 +131,7 @@ function isDeclaration(
     case 'ArrayPattern': // `const [foo, bar] = baz`
       // Array destructuring is truly a declaration on the left side
       // (even if there's reassignment)
-      return parent.elements.includes(id);
+      return id.type === 'Identifier' && parent.elements.includes(id);
```

Comment:
Why do you need to check for the node type here?
Shouldn't this also work for JSX?

Reply:
`parent.elements` is of type `Identifier[]` unfortunately, so we get a type error if we don't check first